packages/flame/lib/src/components/mixins/has_game_ref.dart

```diff
@@ -1,27 +1,31 @@
 import '../../../components.dart';
 import '../../../game.dart';
 
-mixin HasGameRef<T extends BaseGame> on Component {
+mixin HasGameRef<T extends Game> on Component {
   T? _gameRef;
 
   T get gameRef {
-    final ref = _gameRef;
-    if (ref == null) {
-      throw 'Accessing gameRef before the component was added to the game!';
+    if (_gameRef == null) {
+      var c = parent;
+      while (c != null) {
+        if (c is HasGameRef<T> && c._gameRef != null) {
+          _gameRef = c._gameRef;
```

Comment:
Even easier:
```suggestion
        if (c is HasGameRef<T>) {
          _gameRef = c.gameRef;
```
That way `_gameRef` will be cached in the closest ancestor of the current component.